solutions/fully-configurable/main.tf

```diff
@@ -264,6 +264,66 @@ module "vsi" {
   monitoring_agent_version         = var.monitoring_agent_version
 }
 
+########################################################################################################################
+# Trusted Profile for Logging Agent
+########################################################################################################################
+
+locals {
+  # Determine if we need to create a trusted profile
+  create_logging_trusted_profile = var.install_logging_agent && var.logging_auth_mode == "VSITrustedProfile" && var.logging_trusted_profile_id == null
+
+  # Extract Cloud Logs instance ID from ingress endpoint
+  cloud_logs_instance_id = var.logging_target_host != null ? element(split(".", var.logging_target_host), 0) : null
+}
+
+module "trusted_profile" {
+  count   = local.create_logging_trusted_profile ? 1 : 0
+  source  = "terraform-ibm-modules/trusted-profile/ibm"
+  version = "3.2.0"
+
+  trusted_profile_name        = "${local.prefix}-vsi-logging-trusted-profile"
+  trusted_profile_description = "Trusted profile for VSI instances to send logs to IBM Cloud Logs"
```

Comment:
I would include the Cloud Logs instance ID in this description since the TP is scoped to that exact instance